torch/profiler/profiler.py

```diff
@@ -219,19 +379,28 @@ def __init__(
             with_modules: bool = False,
             # deprecated:
             use_cuda: Optional[bool] = None):
+
         if activities:
-            self.activities = set(activities)
+            activities_set = set(activities)
         else:
-            self.activities = supported_activities()
+            activities_set = supported_activities()
+        assert len(activities_set) > 0, "No valid profiler activities found"
```

Comment:
Doesn't this assert need to be later, in case no activities are passed but `use_cuda=True`?

Reply:
good catch